exonum/src/runtime/mod.rs

```diff
@@ -198,37 +198,42 @@ pub trait Runtime: Send + Debug + 'static {
     /// * Ensure that the deployed artifact has the following information, even if it is empty.
     fn artifact_protobuf_spec(&self, id: &ArtifactId) -> Option<ArtifactProtobufSpec>;
 
-    /// Restart previously added service instance with the given specification.
+    /// Runs the constructor of a new service instance with the given specification
+    /// and initial configuration.
+    ///
+    /// The service is not guaranteed to be added to the runtime at this point, so the runtime
+    /// may freely discard the instantiated service instance after completing this method. To
```

Comment:
Why `may` is used here? Is implementation of Runtime possible, which performs service instantiation in `start_adding_service` and does nothing in `start_service`?

Reply:
What instantiation of the service leads to changes in the database state? Should these changes be reverted in case of error pr panic?